### DISCUSSION TITLE
Fail to custom power plan with each mode

### DISCUSSION THREAD

**Discussion by ZhennanRu** [seerge/g-helper#356] [Q&A] [answered]
Hello,

I am trying to combine power plan with each mode, following the instruction of "Power User Setting", .

I find the config.json file under "AppData/Roaming/GHelper". But it seems that GHelper will initiate config file every time after I change the power mode. Even when I didn't do any change to the config file, I see the file modified time change.

Computer: ROG Zephyrus M16 (2023) GU604VZ
System: Windows 11 pro 22H2
GHelper version: 0.58.0.0

Thanks

**Comment by seerge** [comment]
@ZhennanRu Most probably you are adding extra comma in the end or so, post your config here.

**Reply by ZhennanRu** [reply]
Thank you for the reply.
I checked that again but still didn't work. Also VSCode didn't report any error.
Here is my config
```
{
  "performance_mode": 2,
  "performance_0": 2,
  "gpu_mode": 0,
  "miniled": 1,
  "frequency": 60,
  "overdrive": 0,
  "performance_1": 0,
  "gpu_auto": 1,
  "screen_auto": 1,
  "keyboard_auto": 1,
  "aura_mode": 2,
  "charge_limit": 95,
  "mid_fan": 1,
  "fan_profile_mid_2": "3C-41-42-46-47-4B-4C-62-03-0C-0C-16-16-22-22-29",
  "fan_profile_cpu_2": "3C-41-42-46-47-4B-4C-62-03-0C-0C-16-16-22-22-29",
  "fan_profile_gpu_2": "3C-41-42-46-47-4B-4C-62-08-11-11-1D-1D-26-26-2D",
  "limit_total_2": 125,
  "limit_cpu_2": 80,
  "keyboard_awake": 1,
  "keyboard_boot": 1,
  "keyboard_sleep": 0,
  "keyboard_shutdown": 1,
  "fan_profile_mid_0": "3A-3D-40-44-48-4D-51-62-08-11-16-1A-22-29-30-45",
  "fan_profile_cpu_0": "3A-3D-40-44-48-4D-51-62-08-11-16-1A-22-29-30-45",
  "fan_profile_gpu_0": "3A-3D-40-44-48-4D-51-62-0C-16-1D-1F-26-2D-34-4A",
  "limit_total_0": 125,
  "limit_cpu_0": 80,
  "scheme_0": "2ac1d0e0-17a7-44ed-8091-d88ef75a4eb0",
  "scheme_1": "381b4222-f694-41f0-9685-ff5bb260df2e"
}
```

**Reply by seerge** [reply]
@ZhennanRu i just checked, config is fine and also works fine in app? App sets your power plan IDs for Balanced and Turbo ids ? what's wrong?

Fact that app keeps modifying own config, is absolutely fine, as it doesn't erase schemes from it.

**Reply by ZhennanRu** [reply]
Here is the video recording. I copy paste the config code and saved. But the last two scheme objects directly disappeared when I switch power mode.

https://github.com/seerge/g-helper/assets/73455942/eb1eb078-9178-4e06-88cc-49d3c1712303

**Reply by seerge** [reply, marked as answer]
@ZhennanRu you need to quit app before editing, it reads config only on start, and then only writes own updates.

**Reply by ZhennanRu** [reply]
Oh I see, it's resolved now.
Thank you very much!
